src/util/misc.js

```diff
@@ -290,8 +290,147 @@
       );
     },
 
+    /**
+     * Returns the transform matrix according to `relationToObject`.\
+     * The returned matrix is relative to the entire object plane spread by a canvas,
+     * meaning it doesn't account for canvas viewport transform.
+     *
+     * `child` relation means `point` exists in the coordinate plane created by the object.
+     * In other words, point is measured acoording to object's center point
+     * meaning that if `point` is equal to (0,0) it is positioned at object's center.\
+     * `sibling` relation means `point` exists in the same coordinate plane as object.
+     * In other words they both relate to the same (0,0) and agree on every point.
+     *
+     * @static
+     * @memberOf fabric.util
+     * @param {fabric.Object} object
+     * @param {'sibling'|'child'} relationToObject
+     * @returns {number[]} plane matrix relative to the coordinate plane created by a canvas
+     */
+    getTransformMatrixByObject: function (object, relationToObject) {
+      if (relationToObject !== 'child' && relationToObject !== 'sibling') {
+        throw new Error('fabric.js: recieved bad argument ' + relationToObject);
+      }
+      return relationToObject === 'child' ?
+        object.calcTransformMatrix() :
+        object.group ?
+          object.group.calcTransformMatrix() :
+          fabric.iMatrix.concat();
+    },
+
+
+    /**
+     * Calculate the transform matrix needed to apply to target (object/point/whatever)
+     * in order for it to be sent to the destination coordinate plane **without** being changed from the canvas/viewer's perspective.
+     * In other words, target exists on the source plane and we need it to be moved to the destination plane
+     * but we want it to remain unchanged from our perspective.
+     * This method will return the transformation that needs to be added to target (pre-transform).
+     *
+     * `child` relation means target exists in the coordinate plane created by the object we relate to.
+     * In other words:
+     * - target is measured acoording to object's center point
+     * - target is drawn by object in the plane object creates
+     * - target is transformed by object but is unaware of it.
+     * - if target is positioned at (0,0) it is positioned at object's center.
+     *
+     * `sibling` relation means target exists in the same coordinate plane as the object we relate to.
+     * In other words they both relate to the same (0,0), agree on every point and are drawn in the same transformed plane.
```

Comment:
We could add an explanation for simple people like me, making the example that you want to move one object from the canvas inside a group, or from a group to another, but without changing its perceived position and properties